### DISCUSSION TITLE
Adapting the Sensel Morph for steno

### DISCUSSION THREAD

**Discussion by panathea** [opensteno/plover#1330] [Show and tell]
I've had a [Sensel Morph](https://morph.sensel.com/) for a couple years and I've been trying occasionally to get it working with Plover. It would be a budget-friendly ($300 USD) alternative to other pressure-sensitive machines which currently run for upwards of $2000 USD.

### Advantages

- "Affordable"
- Customizable
- Can act as a trackpad simultaneously
- Directly readable by Plover over MIDI
- Bluetooth and USB connectivity

### Limitations

- **Blocker** no ability to send a single zone as multiple MIDI notes (e.g. hitting the crack between T and K cannot send "TK"
- The sensitivity seems to be practically around 5g, which can feel not-sensitive-enough (at least for me compared to the lightspeed?)
- Creating a custom overlay seems to be a largely undeveloped space, so we have to:
    - Work directly on the surface of the morph
    - Use an innovator's overlay (reduces sensitivity in my experience)
    - Print a custom overlay (haven't tried this)

### Current attempt

The latest attempt I've had was with the [laser-cut acrylic keytoppers from the Plover Store.](http://plover.deco-craft.com/shop/view_product/Laser-Cut-Steno-Keys-Kit)

The thinking was that by having the keys on sticky pads, we'd be able to dissipate single-finger force into two pressure zones, thereby working around the software limitation of only sending one note per pressure point:

![image](https://user-images.githubusercontent.com/5840970/119164698-ae43be00-ba2a-11eb-8d63-9ce57dc7f08a.png)

However, the result when pressing the cracks even with the sticky pads 5mm apart is that they are registered largely as one point:

![image](https://user-images.githubusercontent.com/5840970/119164898-e5b26a80-ba2a-11eb-9435-ce7de6e2cc3d.png)

### Next steps

Let Sensel know we need this feature. I know that sillabix on Discord has reached out to them in 2019 over email, but the more noise we make the higher priority we will be. Please start a new post on the Sensel Forum or [comment on my post.](https://forum.sensel.com/t/feature-request-overlap-zones/2179)

**Comment by user202729** [comment]
... how hard could it be to workaround it? Sounds like you can just make 3 zones...

![a](https://user-images.githubusercontent.com/25191436/119209358-58295600-bad0-11eb-9fe3-e281b4c71e56.png)

Or is it that the keyboard can't send the full details (pressure+location+area?), only the translated notes/centroid of the pressed region?

Unless the board detects no difference between 1 and 2-key pressed down, then it's a hardware limitation.

**Comment by panathea** [comment]
The software doesn't support one point sending multiple commands, or two
points being hit at once.

Using the raw data to make the machine plugin is disadvantageous because
then it only works over a wired connection.

**Reply by user202729** [reply]
(I don't really understand how it works exactly, or I have that machine.)
